### PR TITLE
Clients (test and bench) refactoring

### PR DESCRIPTION
**Pull request by jzuniga-amd**

### CODE REVIEW

File: rocblascommon/clients/include/d_vector.hpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            U host[PAD];
+
+            // Copy device memory after allocated memory to host
+            hipMemcpy(host, d + this->size, sizeof(guard), hipMemcpyDeviceToHost);
+
+            // Make sure no corruption has occurred
+            EXPECT_EQ(memcmp(host, guard, sizeof(guard)), 0);
+
+            // Point to guard before allocated memory
+            d -= PAD;
```
Comment (tfalders) [inline rocblascommon/clients/include/d_vector.hpp:91]: Should there be a d += PAD at the end to restore d?

Comment (jzuniga-amd) [reply rocblascommon/clients/include/d_vector.hpp:91]: padded vectors have the form [PAD1 data PAD2], and a pointer d_vector points to the first element in data, so line 85 with d+size is reading PAD2, while line 94 with d-PAD is reading PAD1...
The comment in line 93 is wrong though... it should be "copy device memory **before** allocated memory to host"

File: rocsolver/clients/gtest/larfb_gtest.cpp [hunk 2]
```diff
@@ -19,32 +14,38 @@ using namespace std;
...
 //if d = 0, then direct = 'F'
 //if d = 1, then direct = 'B'
 //if t = 0, then trans = 'N'
 //if t = 1, then trans = 'T'
 //if t = 2, then trans = 'C'
+
+// case when m = 0 and k = 0 will also execute the bad arguments test
+// (null handle, null pointers and invalid values)
+
+const vector<vector<int>> matrix_size_range = {
```
Comment (tfalders) [inline rocsolver/clients/gtest/larfb_gtest.cpp:33]: Missing comments "for daily_lapack/checkin_lapack tests"

File: rocsolver/clients/gtest/larfb_gtest.cpp [hunk 3]
```diff
@@ -19,32 +14,38 @@ using namespace std;
 
 typedef std::tuple<vector<int>, vector<int>> bTuple;
 
-//{M,N,lda,s,ldv,st}
+// each matrtix_size vector is a {M,N,lda,s,ldv,st}
```
Comment (tfalders) [inline rocsolver/clients/gtest/larfb_gtest.cpp:17]: Matrix is misspelled

File: rocsolver/clients/gtest/larfg_gtest.cpp [hunk 4]
```diff
@@ -19,115 +14,91 @@ using namespace std;
...
 const vector<int> incx_range = {
-    -1, 0, 1, 5, 8, 10,
+    -1, 0,          //invalid
+     1, 5, 8, 10,
+};
+
+// for checkin_lapack tests
+const vector<int> n_size_range = {
+    0,  //quick return
+    -1, //invlaid
```
Comment (tfalders) [inline rocsolver/clients/gtest/larfg_gtest.cpp:28]: Invalid misspelled

File: rocsolver/clients/gtest/ormlx_unmlx_gtest.cpp [hunk 5]
```diff
@@ -37,10 +28,24 @@ const vector<vector<int>> size_range = {
 // if t = 0, then trans = 'N'
 // if t = 1, then trans = 'T'
 // if t = 2, then trans = 'C'
+
+// case when
```
[the remark this answers is not in the document]
Comment (jzuniga-amd) [reply rocsolver/clients/gtest/ormlx_unmlx_gtest.cpp:32]: fixed

File: rocsolver/clients/common/cblas_interface.cpp [hunk 6]
```diff
@@ -635,87 +651,68 @@ void cblas_orgl2_ungl2<rocblas_double_complex>(rocblas_int m, rocblas_int n, roc
...
-void cblas_ormbr_unmbr<float>(char storev, rocblas_side side, rocblas_operation trans, rocblas_int m, rocblas_int n, rocblas_int k, float *A,
+void cblas_ormbr_unmbr<float>(rocblas_storev storev, rocblas_side side, rocblas_operation trans, rocblas_int m, rocblas_int n, rocblas_int k, float *A,
                                rocblas_int lda, float *ipiv, float *C, rocblas_int ldc, float *work, rocblas_int lwork) {
   int info;
-  char sideC = 'R', transC = 'T';
+  char sideC = rocblas2char_side(side);
+  char transC = rocblas2char_operation(trans);
   char vect;
-  if (storev == 'C')
+  if (storev == rocblas_column_wise)
```
[the remark this answers is not in the document]
Comment (jzuniga-amd) [reply rocsolver/clients/common/cblas_interface.cpp:660]: legacy lapack use here a different type variable (vect) whose values are 'Q' or 'P'. This is equivalent to column_wise and row_wise so I re-used storev, but its values cannot be then transformed into 'C' or 'R' as rocblas2char_storev does.

File: rocsolver/clients/gtest/larfb_gtest.cpp (hunk 2 above)
Comment (jzuniga-amd) [reply rocsolver/clients/gtest/larfb_gtest.cpp:33]: fixed

File: rocsolver/clients/gtest/larfb_gtest.cpp (hunk 3 above)
Comment (jzuniga-amd) [reply rocsolver/clients/gtest/larfb_gtest.cpp:17]: fixed

File: rocsolver/clients/gtest/larfg_gtest.cpp (hunk 4 above)
Comment (jzuniga-amd) [reply rocsolver/clients/gtest/larfg_gtest.cpp:28]: fixed

File: rocsolver/clients/include/testing_orgbr_ungbr.hpp [hunk 7]
```diff
@@ -2,202 +2,263 @@
...
+    // collect performance data 
+    if (argus.timing) 
+        orgbr_ungbr_getPerfData<T>(handle, storev, m, n, k, dA, lda, dIpiv,
+                          hA, hIpiv, &gpu_time_used, &cpu_time_used, hot_calls); 
+        
+    // validate results for rocsolver-test
+    // using s * machine_precision as tolerance
+    rocblas_int s = row ? n : m;
+    if (argus.unit_check) 
+        rocsolver_test_check<T>(max_error,k);
```
[the remark this answers is not in the document]
Comment (jzuniga-amd) [reply rocsolver/clients/include/testing_orgbr_ungbr.hpp:239]: yes... I forgot to update the norm check in this and other tests. All of them should be fixed now.

**Pull request merged**